backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -165,25 +157,83 @@ public ProjectDAORdbImpl(
     this.starterProjectID = app.getStarterProjectID();
   }
 
-  private void checkIfEntityAlreadyExists(Session session, Project project) {
-    ModelDBHibernateUtil.checkIfEntityAlreadyExists(
-        session,
-        "p",
-        GET_PROJECT_COUNT_BY_NAME_PREFIX_HQL,
-        "Project",
-        "projectName",
-        project.getName(),
-        ModelDBConstants.WORKSPACE,
-        project.getWorkspaceId(),
-        project.getWorkspaceType(),
-        LOGGER);
+  private void checkIfEntityAlreadyExists(
+      Session session, Workspace workspace, String projectName) {
+    List<String> projectEntityIds = getProjectIdsByName(session, projectName);
```

Comment:
Project names only have to be unique in the workspace. We can have duplicated names in different workspaces. I believe this code is checking for globally unique, right?

Reply:
As per the old logic, we have a unique project name in the workspace so currently I am checking uniqueness at the workspace level. We can have duplicate names in different workspaces.https://github.com/VertaAI/modeldb/pull/1795/files#diff-19df052e74b46634ca35a406234e4539b5f4e5ea7a7c18f743fc8162ba1847c2R615-R635